src/raftstore/store/fsm/peer.rs

```diff
@@ -361,6 +361,11 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
                 self.fsm.group_state = GroupState::Chaos;
                 self.register_raft_base_tick();
             }
+            CasualMessage::SnapshotGenerated => {
+                // Resume snapshot handling again to avoid waiting another heartbeat.
+                self.fsm.peer.ping();
+                self.fsm.has_ready = true;
```

Comment:
Maybe a follower still has to wait for the next heartbeat to request snapshots because it may replies this ping a little later.
So how about make has_ready become `RefCell<bool>` and set true when a snapshot is needed in PeerStorage::snapshot?

Reply:
> Maybe a follower still has to wait for the next heartbeat to request snapshots because it may replies this ping a little later.

That's OK. Setting `has_ready` to true is just for sending out the heartbeats. Snapshot is generated, there is nothing more for leader to do except confirming follower still needs the snapshot.

Note that `ping` here does not provide a guarantee, it's just trying its best effort.